packages/networking/lib/connect.ts

```diff
@@ -0,0 +1,35 @@
+import * as net from 'net'
+import * as dns from 'dns'
+import * as Promise from 'bluebird'
+
+export function byPortAndAddress (port: number, address: net.Address) {
+  // https://nodejs.org/api/net.html#net_net_connect_port_host_connectlistener
+  return new Promise((resolve, reject) => {
+    const client = net.connect(port, address.address, () => {
+      client.end()
```

Comment:
we should probably remove the `error` listener here too once the connect resolves - although it would eventually get GC'd when it falls out of scope.